torch/utils/_pytree.py

```diff
@@ -282,11 +309,44 @@ def _private_register_pytree_node(
         SUPPORTED_SERIALIZED_TYPES[cls] = serialize_node_def
         SERIALIZED_TYPE_TO_PYTHON_TYPE[serialized_type_name] = cls
 
+        if flatten_with_keys_fn is not None:
+            SUPPORTED_KEY_PATH_NODES[cls] = _KeyPathNodeDef(cls, flatten_with_keys_fn)
+
+
+@dataclasses.dataclass(frozen=True)
+class SequenceKey:
```

Comment:
nit/bikeshed: IndexKey (more correct) or GetItemKey (matches GetAttrKey) or ListIndex/Key (Matches DictKey)

Reply:
I changed this to map to the collections.abc scheme (SequenceKey, MappingKey). GetAttrKey keeps its name because there is no abstract base class that represents "bag of attributes" and I couldn't think of a better name.